apps/etterlatte-beregning/src/main/kotlin/beregning/grunnlag/BeregningsGrunnlagRoutes.kt

```diff
@@ -7,41 +7,59 @@ import io.ktor.server.response.respond
 import io.ktor.server.routing.Route
 import io.ktor.server.routing.get
 import io.ktor.server.routing.post
+import io.ktor.server.routing.route
 import no.nav.etterlatte.beregning.klienter.BehandlingKlient
 import no.nav.etterlatte.libs.common.BEHANDLINGSID_CALL_PARAMETER
 import no.nav.etterlatte.libs.common.withBehandlingId
 import no.nav.etterlatte.libs.ktor.bruker
 import org.slf4j.LoggerFactory
+import java.util.*
 
 private val logger = LoggerFactory.getLogger("BeregningsGrunnlag Route")
 
 fun Route.beregningsGrunnlag(beregningsGrunnlagService: BeregningsGrunnlagService, behandlingKlient: BehandlingKlient) {
-    post("/api/beregning/beregningsgrunnlag/{$BEHANDLINGSID_CALL_PARAMETER}/barnepensjon") {
-        withBehandlingId(behandlingKlient) { behandlingId ->
-            val body = call.receive<BarnepensjonBeregningsGrunnlag>()
-
-            when {
-                beregningsGrunnlagService.lagreBarnepensjonBeregningsGrunnlag(
-                    behandlingId,
-                    body,
-                    bruker
-                ) -> call.respond(HttpStatusCode.NoContent)
-
-                else -> call.respond(HttpStatusCode.Conflict)
+    route("/api/beregning/beregningsgrunnlag") {
+        post("/{$BEHANDLINGSID_CALL_PARAMETER}/fra/{forrigeBehandlingId}") {
+            withBehandlingId(behandlingKlient) { behandlingId ->
+                val forrigeBehandlingId = call.parameters["forrigeBehandlingId"]?.let {
+                    UUID.fromString(it)
+                } ?: throw NullPointerException(
+                    "Forrige BehandlingsId er ikke i path params"
+                )
+
+                beregningsGrunnlagService.dupliserBeregningsGrunnlag(behandlingId, forrigeBehandlingId)
+
+                call.respond(HttpStatusCode.NoContent)
+            }
```

Comment:
Denne ruten bør kun brukes av andre apper, og ikke direkte fra frontend. Jeg vet ikke om vi har noe konsept om slik tilgangsstyring enda. Hvis vi har noe lett tilgjengelig som sperrer for saksbehandler-kall ville jeg lagt det på her.

Reply:
Fjernet withBehandlingId (siden den sjekker for saksbehandler har tilgang) og endret det til at det er kun SystemBruker som kan kalle.

Passer det?